app/src/main/java/com/github/h3lp3rs/h3lp/signin/SignInActivity.kt

```diff
@@ -70,10 +84,24 @@ class SignInActivity : AppCompatActivity() {
         signInClient.authenticate(result, activity)
             ?.addOnCompleteListener(activity) { task ->
                 if (task.isSuccessful) {
-                    // Sign in success, update activity with the signed-in user's information
-                    val intent = Intent(activity, MainPageActivity::class.java)
-                    startActivity(intent)
+                    checkToSAndLaunchIfNotAcceptedElseMain()
                 }
             }
     }
+
+    /**
+     * Companion used to pass on a global context used to open the local
+     * storages
+     */
+    companion object{
+        @SuppressLint("StaticFieldLeak")
+        lateinit var globalContext: Context
```

Comment:
As discussed, this is not the neatest solution but is quite practical in the context of the local storages (and seems to be recommended on a few sources). We could discuss further to find out if that is the only solution, in which case, a couple of other classes could be simplified by using this global context instead of passing on an activity's context on every call (e.g. LocalEmergencyCall).